File: backend/weave-java/src/main/java/almanac/models/place/Place.java
package almanac.models.place;

public class Place {
    private Long id;
    private String name;
    private String description;
    private PlaceType type;

    public Place(Long id, String name, String description, PlaceType type) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.type = type;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public PlaceType getType() {
        return type;
    }
}

File: backend/weave/src/main/kotlin/almanac/models/Place.kt
package almanac.models

data class Place(
        val id: Long,
        val name: String,
        val description: String,
        val type: PlaceType
)

enum class PlaceType {
    GUILD,
    TAVERN,
    SHOP
}

data class SettlementPlace(
        val settlementId: Long,
        val placeId: Long
)